packages/mobile/src/screens/track-screen/TrackScreen.tsx

```diff
@@ -66,19 +67,20 @@ export const TrackScreen = () => {
     FeatureFlags.COMMENTS_ENABLED
   )
 
-  const handleFetchTrack = useCallback(() => {
-    dispatch(tracksActions.reset())
-    dispatch(
-      fetchTrack(
-        id ?? null,
-        decodeURIComponent(slug ?? ''),
-        handle ?? user?.handle,
-        canBeUnlisted
+  const isScreenReady = useIsScreenReady()
+  useEffect(() => {
```

Comment:
good for now! could still maybe do a useIsScreenReadyEffect and use that in screencontext at minmum?

Reply:
note, as andrew said, maybe we add useIsPrimaryContentEffect something rather in the odd case that you have to have some global business logic at the root